EIPS/eip-2274.md

```diff
@@ -0,0 +1,167 @@
+
+## Rationale
+<!--The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages. The rationale may also provide evidence of consensus within the community, and should discuss important objections or concerns raised during discussion.-->
+A few proposals inspired this one. First, [eip-1352](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1352.md) proposes to specify an Ethereum address range occupied by precompiles. As such, we see that a similar approach could be taken to split the address range dedicated to precompiled contracts into a set of addresses to be used for mainnet precompiled contracts and a set of addresses for forks precompiled. This solution has the advantage to be trivial to implement. Nevertheless, it fails to provide a nice level of abstraction, which is what this proposal aims to address. Likewise, [eip-1109](https://eips.ethereum.org/EIPS/eip-1109) and [eip-1702](https://eips.ethereum.org/EIPS/eip-1702) inspired this proposal.
+
+## Backwards Compatibility
+<!--All EIPs that introduce backwards incompatibilities must include a section describing these incompatibilities and their severity. The EIP must explain how the author proposes to deal with these incompatibilities. EIP submissions without a sufficient backwards compatibility treatise may be rejected outright.-->
+This EIP requires a hardfork to be able to extend the set of OPCODES. However, it is backward compatible since no currently deployed smart contracts will be broken if this EIP was to be introduced and implemented.
+
+## Test Cases
+<!--Test cases for an implementation are mandatory for EIPs that are affecting consensus changes. Other EIPs can choose to include links to test cases if applicable.-->
+Same as in [eip-1109](https://eips.ethereum.org/EIPS/eip-1109), with:
+- Call to defined and undefined precompiled for different `vmID` values.
+
+## Implementation
+<!--The implementations must be completed before any EIP is given status "Final", but it need not be completed before the EIP is accepted. While there is merit to the approach of reaching consensus on the specification and rationale before writing code, the principle of "rough consensus and running code" is still useful when it comes to resolving many discussions of API details.-->
+In order to support this new OPCODE and **illustrate** the point made above, we propose an **incomplete and broken** implementation below:
+- In `core/vm/instructions.go` (see: [here](https://github.com/ethereum/go-ethereum/blob/master/core/vm/instructions.go)), add a function for the new OPCODE:
```

Comment:
Putting geth code in an EIP presents licensing problems as it is LGPL licensed and has a copyleft clause which compels derivative works to be placed under LGPL.  That copyleft clause is incompatible with the EIP requirement that all EIPs be public domain works.

Consider either switching to pseudo code or creating a PR and referencing the PR in this EIP.

Reply:
Thanks @shemnon, I removed the code and added links to a fork of geth containing the functions.